src/Middleware/WebSockets/src/HandshakeHelpers.cs

```diff
@@ -43,11 +43,11 @@ public static bool CheckSupportedWebSocketRequest(string method, List<KeyValuePa
                 return false;
             }
 
-            foreach (var pair in headers)
+            foreach (var pair in interestingHeaders)
             {
                 if (string.Equals(HeaderNames.Connection, pair.Key, StringComparison.OrdinalIgnoreCase))
                 {
-                    if (string.Equals(Constants.Headers.ConnectionUpgrade, pair.Value, StringComparison.OrdinalIgnoreCase))
+                    if (string.Equals(HeaderNames.Upgrade, pair.Value, StringComparison.OrdinalIgnoreCase))
```

Comment:
While this is a constant that should never change so technically correct, it feels wrong to repurpose the constant for `Upgrade: ` header name to compare against the value of a potential `Connection: Upgrade` header.

Reply:
Actually thinking about the semantics of the WebSocket handshake more, I'm completely okay with it 😄

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection

`Upgrade` in the `Connection` header; refers to the `Upgrade` header; so it is technically correct to use it?

![image](https://user-images.githubusercontent.com/1142958/112768243-048c1600-9013-11eb-8af8-4649c4a75e92.png)

Reply:
Looks like `Connection` is a list of header names?

https://tools.ietf.org/html/rfc7230#section-6.1

> When a header field aside from Connection is used to supply control
   information for or about the current connection, the sender MUST list
   the corresponding field-name within the Connection header field.  A
   proxy or gateway MUST parse a received Connection header field before
   a message is forwarded and, for each connection-option in this field,
   remove any header field(s) from the message with the same name as the
   connection-option, and then remove the Connection header field itself
   (or replace it with the intermediary's own connection options for the
   forwarded message).
>
>   Hence, the Connection header field provides a declarative way of
   distinguishing header fields that are only intended for the immediate
   recipient ("hop-by-hop") from those fields that are intended for all
   recipients on the chain ("end-to-end"), enabling the message to be
   self-descriptive and allowing future connection-specific extensions
   to be deployed without fear that they will be blindly forwarded by
   older intermediaries.